plutus-ir/src/Language/PlutusIR/Transform/ThunkRecursions.hs

```diff
@@ -108,6 +111,9 @@ left with the PLC subset. This would be quite elegant - except that the final "l
 would be quite ugly as it would just have to fail if it saw any PIR features.
 -}
 
+thunkRecursionsTerm :: MonadQuote m => Term TyName Name a -> m (Term TyName Name a)
+thunkRecursionsTerm = PLC.rename >=> thunkRecursionsTermGo
```

Comment:
@michaelpj, this looks ugly. I'm also not sure if it's needed at all. I can experiment, but I guess it's better if you just tell me.